### PR TITLE
Fixed a condition where JetStream assets could be created in multiple leafnode domains.

### PR DESCRIPTION
**Pull request by derekcollison**
Also added in optional Domain to StreamInfo.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/leafnode.go [hunk 1]
```diff
@@ -1076,11 +1076,15 @@ func (c *client) processLeafnodeInfo(info *Info) {
 					// account also has JetStream enabled.
 					if accHasJS {
 						s.addInJSDenyExport(remote)
+						// If we specified a domain do not import by default.
+						if hasJSDomain {
+							s.addInJSDenyImport(remote)
+						}
```
Comment (matthiashanel) [inline server/leafnode.go:1082]: This works, but I have one question about this. Why is it that this is guarded by `accHasJs`? 

If a downstream server (say through another leaf node hop) has no JS enabled, adding a deny should not matter as there is no interest on that end.
If such an account does have JS enabled, then there, presumably we would add the deny. Meaning a deny earlier on shouldn't matter either. 

What am I missing?

**Review by matthiashanel — APPROVED** [review]
LGTM, but one question.

File: server/leafnode.go (hunk 1 above)
Comment (derekcollison) [reply server/leafnode.go:1082]: We don't need the deny unless the account has JS enabled (not the server), so just trying to keep it small (the deny stuff).

Not sure that answers your question though..

**Review by kozlovic — COMMENTED** [review]
Will defer to Matthias that knows way more than me on that subject...

**Pull request merged**